src/corehost/cli/fxr/sdk_info.cpp

```diff
@@ -7,9 +7,38 @@
 #include "trace.h"
 #include "utils.h"
 
-bool compare_by_version(const sdk_info &a, const sdk_info &b)
+bool compare_by_version_ascending_then_hive_depth_descending(const sdk_info &a, const sdk_info &b)
 {
-    return a.version < b.version;
+    if (a.version < b.version)
+    {
+        return true;
+    }
+
+    // With multi-level lookup enabled, it is possible to find two SDKs with
+    // the same version. For that edge case, we make the ordering put SDKs
+    // from farther away (global location) hives earlier than closer ones
+    // (current dotnet exe location). Without this tie-breaker, the ordering
+    // would be non-deterministic.
+    //
+    // Furthermore,  nearer earlier than farther is so that the MSBuild resolver
+    // can do a linear search from the end of the list to the front to find the
+    // best compatible SDK.
+    //
+    // Example:
+    //    * dotnet dir has version 4.0, 5.0, 6.0
+    //    * global dir has 5.0
+    //    * 6.0 is incompatible with calling msbuild
+    //    * 5.0 is compatible with calling msbuild
+    //
+    // MSBuild should select 5.0 from dotnet dir (matching probe order) in muxer
+    // and not 5.0 from global dir.
+    if (a.version == b.version)
+    {
+        assert(a.hive_depth != b.hive_depth);
```

Comment:
We're crashing on mac somewhere in here and it may be this assert. I need to remember to bring in my mac to debug.

Reply:
Fixed. Turns out std::sort on Mac is comparing elements with themselves sometimes. That's legal, so I just removed the invalid assert.